pandas/tests/extension/base/io.py

```diff
@@ -0,0 +1,24 @@
+import numpy as np
+import pytest
+
+from pandas.compat import StringIO
+
+import pandas as pd
+import pandas.testing as tm
+
+from .base import BaseExtensionTests
+
+
+class BaseParsingTests(BaseExtensionTests):
+
+    @pytest.mark.parametrize('engine', ['c', 'python'])
+    def test_EA_types(self, engine, data):
+        df = pd.DataFrame({
+            'with_dtype': pd.Series(data, dtype=str(data.dtype))
+        })
+        csv_output = df.to_csv(index=False, na_rep=np.nan)
+        result = pd.read_csv(StringIO(csv_output), dtype={
+            'with_dtype': str(data.dtype)
+        }, engine=engine)
+        expected = df
+        tm.assert_frame_equal(result, expected)
```

Comment:
Hmm...I really would want to somehow take this test and plant it in the `parsers` tests.

That being said, it's nicely integrated with the existing extension tests that I'm not sure it's worth it...

Perhaps we should add a small test with EA types in the `parsers/common.py` ?

Reply:
Honestly I don't think it would add much value. This feature needs to be tested generically. When I was developing this feature had a small test that I was using but it was too specific to really add any value and didn't cover any use cases that we not already covered by the existing test suite. 

I can add a small test there if you still want, but IMHO it won't add that much value.

Reply:
yeah this is ok here; @gfyoung  wouldn't object to *also* having a test in parsers/common.py that maybe reads in a sample with a bunch of EA types (thinking Decimal and Int64), integrated with the existing tests. but can be a followup.

Reply:
> wouldn't object to also having a test in parsers/common.py that maybe reads in a sample with a bunch of EA types (thinking Decimal and Int64), integrated with the existing tests. but can be a followup.

@jreback : Yep, that's what I had in mind.

Reply:
thanks @gfyoung